exec.go

```diff
@@ -91,7 +108,118 @@ EXAMPLE:
 		},
 	},
 	Action: func(context *cli.Context) error {
-		// TODO
-		return nil
+		params, err := validExecParams(context)
+		if err != nil {
+			return err
+		}
+
+		return execute(params)
 	},
 }
+
+func validExecParams(context *cli.Context) (execParams, error) {
```

Comment:
This name is a bit confusing - initially I thought it was going to validate the parameters, but it's actually generating them. Hence, maybe something like `generateExecParams()` would be clearer.